metrics/internal/cache/ceph-blocklist.go

```diff
@@ -117,13 +119,18 @@ func (c *CephBlocklistStore) Resync() error {
 
 	blockList, err := runCephOSDBlocklist(&c.monitorConfig)
 	if err != nil {
-		return fmt.Errorf("failed to get blocklist data: %v", err)
+		klog.Errorf("failed to get blocklist data: %v", err)
+		if err == context.DeadlineExceeded {
+			c.monitorConfig, err = initCeph(c.kubeClient, c.allowedNamespaces)
+			if err != nil {
+				return fmt.Errorf("failed to initialize ceph: %v", err)
```

Comment:
```suggestion
				return fmt.Errorf("failed to initialize ceph: %v", err)
```
We are not initializing ceph. We are re-creating client.